great_expectations/data_context/data_context/base_data_context.py

```diff
@@ -1913,6 +1913,22 @@ def _instantiate_datasource_from_config(
             )
         return datasource
 
+    def update_datasource(
```

Comment:
It feels like there is more to do here, and that maybe we don't want to pass a config object here but a datasource one. If someone grabs a context with a datasource, makes changes, how do they save it? Other methods work to maintain the cached datasource collection, is that same effort needed here?

Reply:
I agree! I've modified the signature to take in an actual datasource.

Additionally, I've made it such that `BaseDataContext` doesn't persist by default and `DataContext` does. The cache has also been updated.

Let me know if this is closer to what you're envisioning.